demos/embeddings/watch-your-step-cora-demo.ipynb

```diff
@@ -0,0 +1,449 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data_dir = os.path.expanduser(\"~/data/cora\")\n",
+    "os.listdir(data_dir)\n",
+    "edgelist = pd.read_csv(os.path.join(data_dir, \"cora.cites\"), sep='\\t', header=None, names=[\"target\", \"source\"])\n",
+    "edgelist[\"label\"] = \"cites\"\n",
+    "\n",
+    "Gnx = nx.from_pandas_edgelist(edgelist, edge_attr=\"label\")\n",
+    "nx.set_node_attributes(Gnx, \"paper\", \"label\")\n",
+    "\n",
+    "feature_names = [\"w_{}\".format(ii) for ii in range(1433)]\n",
+    "column_names =  feature_names + [\"subject\"]\n",
+    "node_data = pd.read_csv(os.path.join(data_dir, \"cora.content\"), sep='\\t', header=None, names=column_names)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Creating the model\n",
+    "\n",
+    "First, we create a StellarGraph object from our NetworkX graph. From this, we create an `AdjacencyPowerGenerator` which loops through the rows of the first `num_powers` of the adjacency matrix."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/Users/kieranricardo/PycharmProjects/origin-stellargaph/stellargraph/stellargraph/core/experimental.py:88: ExperimentalWarning: AdjacencyPowerGenerator is experimental: lack of unit tests. It may be difficult to use and may have major changes at any time.\n",
+      "  warnings.warn(direct_msg, ExperimentalWarning)\n"
+     ]
+    }
+   ],
+   "source": [
+    "G = StellarGraph(Gnx)\n",
+    "generator = AdjacencyPowerGenerator(G, num_powers=10)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, we use the `WatchYourStep` class to create trainable node embeddings and expected random walks."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/Users/kieranricardo/PycharmProjects/origin-stellargaph/stellargraph/stellargraph/core/experimental.py:88: ExperimentalWarning: WatchYourStep is experimental: lack of unit tests. It may be difficult to use and may have major changes at any time.\n",
+      "  warnings.warn(direct_msg, ExperimentalWarning)\n",
+      "/Users/kieranricardo/PycharmProjects/origin-stellargaph/stellargraph/stellargraph/layer/watch_your_step.py:147: UserWarning: embedding_dimension must be even. embedding_dimension will be rounded down to the nearest even number.\n",
+      "  \"embedding_dimension must be even. embedding_dimension will be rounded down to the nearest even number.\"\n",
+      "/Users/kieranricardo/PycharmProjects/origin-stellargaph/stellargraph/stellargraph/core/experimental.py:88: ExperimentalWarning: AttentiveWalk is experimental: lack of unit tests. It may be difficult to use and may have major changes at any time.\n",
```

Comment:
Ah, hm, is `AttentiveWalk` purely an implementation detail of WyS? If so, and a user should never call it directly, it might be better to not mark it `@experimental`, so that there's not this warning. If a user might call it directly, I guess it's probably ok.

Reply:
yeah good point, I don't think a user would ever call this directly